typed-racket-test/succeed/pr241-variation-5.rkt

```diff
@@ -0,0 +1,339 @@
+ [TCP-Listener (tcp-listen 0) choice-evt]
+ [Variable-Reference (let ([x 4]) (#%variable-reference x)) variable-reference->empty-namespace]
+ [Impersonator-Property (let-values ([(i i? i-val) (make-impersonator-property 'i)]) i) (lambda (i) (impersonate-procedure (lambda () (void)) #f i 2))]
+ [Semaphore (make-semaphore) semaphore-post]
+ [FSemaphore (make-fsemaphore 0) fsemaphore-post]
+ [Bytes-Converter (or (bytes-open-converter "UTF-8" "UTF-8") (error 'pr241 "Failed to make bytes converter")) bytes-close-converter]
+ [Pseudo-Random-Generator (current-pseudo-random-generator) pseudo-random-generator->vector]
+ [Logger (current-logger) (lambda (l) (log-level? l 'info))]
+ [Log-Receiver (make-log-receiver (current-logger) 'info) choice-evt]
+ [Log-Level 'info symbol->string]
+ [Place-Channel (let-values ([(p1 p2) (place-channel)]) p1) choice-evt]
+ [Will-Executor (make-will-executor) choice-evt]
+ [FlVector (flvector 1.14 2.14 3.14) flvector-length]
+ [ExtFlVector (extflvector pi.t) extflvector-length]
+ [FxVector (fxvector 1) fxvector-length]
+ [Continuation-Mark-Set (current-continuation-marks) continuation-mark-set->context]
+ [False #f not]
+ [True #t not]
+ [Null '() length]
+))
+
+(define-values-for-syntax (base-untyped* base-typed*)
+  (for/fold ([u* '()] [t* '()])
+            ([tvf (in-list known-base-types)])
+    (unless (and (list? tvf) (= 3 (length tvf)))
+      (error 'pr241 "Expected (TYPE VAL FUN)"))
+    (with-syntax ([type (car tvf)]
+                  [val (cadr tvf)]
+                  [use (caddr tvf)]
+                  [pos-blame 'pr241-test]
+                  [neg-blame 'known-base-types]
+                  [id (gensym (car tvf))])
+      ;; Wrap value in contract, pass it to `use` or assert a wrap failure
+      ;; Ignore the result
+      (define u-stx
+        (if (eq? #f (caddr tvf))
+          #'(with-handlers ([exn:fail:contract? (lambda (e) (void))])
+              (contract any-wrap/c val 'pos-blame 'neg-blame)
+              (error 'pr241 (format "Higher-Order value '~a' incorrectly allowed as Any" val)))
+          #'((lambda any* (void))
+             (use (contract any-wrap/c val 'pos-blame 'neg-blame)))))
+      ;; Bind the value to a typed identifier
+      (define t-stx
+        #'(begin (: id type) (define id val)))
+      (values
+        (cons u-stx u*)
+        (cons t-stx t*)))))
+
+;; `known-polymorphic-types` is a list of 4-lists:
+;;   [TYPE-TAG TYPE VALUE USE]
+;; TYPE-TAG is a polymorphic type exported by Typed Racket
+;; TYPE is an instantiation of TYPE-TAG
+;; VALUE is an expression with type TYPE
+;; USE is the same as for `known-base-types`
+(define-for-syntax known-poly-types '(
+  ;; --- TODO need value for these types
+  ;[Async-Channelof (Async-Channelof Any) (make-async-channel) async-channel-try-get]
+  ;[Prompt-Tagof (Prompt-Tagof Any Any) (default-continuation-prompt-tag) #f]
```

Comment:
Do they have chaperones by default, or do we need to make a chaperone in `any-wrap`?